dvc/dependency/repo.py

```diff
@@ -94,29 +94,25 @@ def dumpd(self, **kwargs) -> dict[str, Union[str, dict[str, str]]]:
         }
 
     def download(self, to: "Output", jobs: Optional[int] = None):
-        from dvc_data.hashfile.build import build
-        from dvc_data.hashfile.checkout import CheckoutError, checkout
-
-        try:
-            repo = self._make_fs(locked=True).repo
-
-            _, _, obj = build(
-                repo.cache.local,
-                self.fs_path,
-                repo.dvcfs,
-                repo.cache.local.fs.PARAM_CHECKSUM,
-            )
-            checkout(
-                to.fs_path,
-                to.fs,
-                obj,
-                self.repo.cache.local,
-                ignore=None,
-                state=self.repo.state,
-                prompt=confirm,
-            )
-        except (CheckoutError, FileNotFoundError):
-            super().download(to=to, jobs=jobs)
+        from dvc.fs import LocalFileSystem
+
+        files = super().download(to=to, jobs=jobs)
+        if not isinstance(to.fs, LocalFileSystem):
+            return files
+
+        hashes: list[tuple[str, HashInfo, dict[str, Any]]] = []
+        for src_path, dest_path in files:
+            try:
+                hash_info = self.fs.info(src_path)["dvc_info"]["entry"].hash_info
+                dest_info = to.fs.info(dest_path)
+            except (OSError, KeyError, AttributeError):
```

Comment:
Why add `OSError`? Were you hitting that in some scenario? I was only trying to cover scenarios where `hash_info` can't be found.

Reply:
I was trying to be defensive here. But you are right, I can remove it.